### PR TITLE
deps: bump go version to 1.22

### PR DESCRIPTION
**Pull request by jmdeal**
Fixes #N/A 

**Description**
Bumps go to the latest release: [release notes](https://tip.golang.org/doc/go1.22)

Justification for pinning to the patch version: https://github.com/golang/go/issues/62278

**How was this change tested?**

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jmdeal** [comment]
Blocked on an apparent issue with govulncheck: https://github.com/golang/go/issues/65590.

**Review by jmdeal — COMMENTED** [review]
/karpenter snapshot

File: test/suites/scale/deprovisioning_test.go [hunk 1]
```diff
@@ -678,7 +678,7 @@ var _ = Describe("Deprovisioning", Label(debug.NoWatch), Label(debug.NoEvents),
 		}, SpecTimeout(time.Hour))
 	})
 	Context("Interruption", func() {
-		It("should interrupt all nodes due to scheduledChange", func(_ context.Context) {
+		FIt("should interrupt all nodes due to scheduledChange", func(_ context.Context) {
```
Comment (engedaam) [inline test/suites/scale/deprovisioning_test.go:681]:
```suggestion
		It("should interrupt all nodes due to scheduledChange", func(_ context.Context) {
```

**Review by engedaam — APPROVED** [review]
LGTM 🚀

**Pull request merged**